include/pipe.h

```diff
@@ -53,13 +53,41 @@ class Pipe {
     darcy_friction_ = darcy_friction;
   }
 
+  //! Assign pipe roughness coefficient
+  //! \param[in] pipe_roughness roughness coefficient of the pipe
+  void pipe_roughness(double pipe_roughness) {
+    pipe_roughness_ = pipe_roughness;
+  }
+
   //! Calculate and return discharge using Darcy-Weisbach equation:
-  //! dhead = (8*darcy_factor*pow(discharge,2)/(pow(M_PI,2)*g*pow(2*radius,5))
+  //! dhead/length =
+  //! (8*darcy_factor*pow(discharge,2)/(pow(M_PI,2)*g*pow(2*radius,5)) That is,
+  //! discharge =
+  //! sqrt(dhead/length*pow(M_PI,2)*g*pow(2*radius,5)/(8*darcy_friction));
+  //! \retval discharge_ discharge in the pipe
+  double discharge_dw();
+
+  //! Calculate and return discharge using Hazen-Williams equation:
+  //! dhead/length =
+  //! (10.67*pow(discharge,1.852)/(pow(pipe_roughness,1.852)*pow(2*radius,4.8704))
```

Comment:
Can I see a reference for these?

Reply:
Headloss section in
https://wntr.readthedocs.io/en/latest/hydraulics.html
And also
https://en.wikipedia.org/wiki/Hazen%E2%80%93Williams_equation